### DISCUSSION TITLE
interactivity api and the router package for multilingual support

### DISCUSSION THREAD

**Discussion by hadimahoor** [WordPress/gutenberg#64774] [Interactivity API]
Greeting to everyone,
First, I would like to thank the interactivity api development team.
As a developer of WordPress themes and Gutenberg block plugins, it has been a great experience.
I hope it will achieve its best conditions.
I wanted to know the opinions of friends about multilingual support in WordPress using the interactivity api and the router package.
Is it easier to implement multilingual support using them?
If so, how do you suggest implementing and using the interactivity api and the router package for multilingual blocks and WordPress themes?
What impact will using them have on SEO?
thanks in advance

**Comment by luisherranz** [comment]
What exactly do you mean by multilingual? That the user can select a language in a drop-down and the page will refresh with the selected language?

**Reply by hadimahoor** [reply]
yes of course
and how interactivity api and router package can improve multilingual experience?

**Reply by luisherranz** [reply]
What kind of improvement are you looking for?

**Reply by hadimahoor** [reply]
Most plugins that add multilingual support to WordPress cause the site to slow down. By using the Interactivity API and the Router package, it's possible to implement multilingual support as a Single Page Application (SPA), which would definitely create a much better user experience. In your opinion, how can we achieve this SPA experience for WordPress multilingual support without using a multilingual plugin and by using at least one custom fields management plugin?

Could you demonstrate a solution for this? For example, if we assign an attribute for each language to a block for translating the block, how would the Interactivity API and the Router package assist in switching between languages? If a small example could be shown, it would be a very helpful and practical demonstration

**Reply by luisherranz** [reply]
I don't believe that simply using an attribute in the blocks and the router of the Interactivity API can achieve that. Translating a site completely is usually much more complex than that, and in fact, that's why it will be treated as its own phase of Gutenberg in the future. It is true that once that phase is carried out, it will be possible to switch languages using the router without having to refresh the page. But the whole UI, server, and database part needs to be done properly.
